cmd/nri-flex/nri-flex.go

```diff
@@ -9,24 +9,28 @@ import (
 	"github.com/newrelic/nri-flex/internal/integration"
 	"github.com/newrelic/nri-flex/internal/load"
 	"github.com/newrelic/nri-flex/internal/outputs"
-	"github.com/sirupsen/logrus"
 )
 
 func main() {
 	load.StartTime = load.MakeTimestamp()
 	integration.SetEnvs()
-	outputs.InfraIntegration()
 
-	if integration.LambdaCheck() {
-		integration.Lambda()
+	if err := outputs.InfraIntegration(); err != nil {
+		load.Logrus.WithError(err).Fatal("flex: failed to initialize integration")
+	}
+
+	if integration.IsLambda() {
+		if err := integration.ValidateLambdaConfig(); err != nil {
+			load.Logrus.WithError(err).Fatal("flex: failed to validate lambda required config")
+		}
+		integration.HandleLambda()
 	} else {
 		// default process
 		integration.SetDefaults()
-		integration.RunFlex("")
+		integration.RunFlex(integration.FlexModeDefault)
 	}
 
-	err := load.Integration.Publish()
-	if err != nil {
-		load.Logrus.WithFields(logrus.Fields{"err": err}).Fatal("flex: unable to publish")
+	if err := load.Integration.Publish(); err != nil {
```

Comment:
I would leave the check for error like it was, instead of putting everything in one line. It's easier to read and it's also how it's done in most all other files